tests/test_file_processing.py

```diff
@@ -106,6 +106,25 @@ def test_process_files(mongo_database, config, testing_files_for_process, testin
     assert samples_collection.count_documents({"RNA ID": "123_B09", "source": "Alderley"}) == 1
 
 
+def test_process_files_one_wrong_format(mongo_database, config, testing_files_for_process, testing_centres):
+    # Test using files in the files/TEST directory
+    # They include a rogue xlsx file dressed as csv file
+
+    _, mongo_database = mongo_database
+    logger = logging.getLogger(__name__)
+
+    centre_config = config.CENTRES[2]
+    centre_config["sftp_root_read"] = "tmp/files"
+    centre = Centre(config, centre_config)
+    centre.process_files()
+
+    imports_collection = get_mongo_collection(mongo_database, COLLECTION_IMPORTS)
+    samples_collection = get_mongo_collection(mongo_database, COLLECTION_SAMPLES)
+    samples_history_collection = get_mongo_collection(mongo_database, COLLECTION_SAMPLES_HISTORY)
+
+    # check that the valid file still gets processed, even though the bad file is in there
+    assert samples_collection.count_documents({"RNA ID": "789_A02", "source": "Test Centre"}) == 1
```

Comment:
Could you also check that the appropriate error has been logged? Something like:
```
assert centre_file.logging_collection.get_count_of_all_errors_and_criticals() == 1
assert centre_file.logging_collection.aggregator_types["TYPE 10"].count_errors == 1
```

Reply:
Thanks, yes will do

Reply:
Done - didn't have access to the CentreFile instance, but can check the imports collection messages.